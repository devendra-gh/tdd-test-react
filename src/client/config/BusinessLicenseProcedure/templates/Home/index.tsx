import React from 'react';
import Container from 'client/containers';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import JourneyLandingTemplate from '@tamm/ui-lib-v2-journey-landing-template';
import Sidebar from 'client/templates/Sidebar';
import { steps } from '../../steps';

/**
 * Home template
 * @param       {Object} props
 * @returns     {JSX}
 */
function Home(props: IVariables) {
  const { i18n } = props;
  return (
    <>
      <Container
        locale={props.locale}
        sidebar={
          <Sidebar
            currentStep={props.currentStep}
            currentSubStep={props.currentSubStep}
            i18n={props.i18n}
            steps={props.steps}
            showRelatedJourneyCard
            stepsStatus={props.stepsStatus}
          />
        }
      >
        <div className="ui-lib-story-container ui-lib-story-container_bg-light-grey">
          <JourneyLandingTemplate
            description={i18n('businessLicenseProcedure.home.description')}
            process={{
              steps: steps.map(step => ({
                description: i18n(`${step.name}.desc`),
                label: i18n(step.name),
              })),
              title: i18n('label.process'),
            }}
            startLogin={{
              buttonLabel: i18n('button.start'),
              description: i18n(
                'businessLicenseProcedure.home.start.description',
              ),
              title: i18n('businessLicenseProcedure.main.title'),
              onClick: () => props.onStart(props),
            }}
            title={i18n('label.serviceDescription')}
            helpfulBlock={{}}
          />
        </div>
        <div style={{ height: 200 }} />
      </Container>
    </>
  );
}

Home.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Home);
