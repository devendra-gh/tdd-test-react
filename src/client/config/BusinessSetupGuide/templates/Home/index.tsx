import { IVariables, withTemplateHooks } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import Container from 'client/containers';
import React from 'react';
import { Link } from 'react-router-dom';
import { ICategory } from '../../data';
import './home.less';

/**
 * Home template
 * @param       {Object} props
 * @returns     {JSX}
 */
function Home(props: IVariables) {
  const { i18n, categoriesAr, categoriesEn } = props;
  const categories = props.locale === 'en' ? categoriesEn : categoriesAr;

  props.actions.breadcrumbs.update(props.breadcrumbs);

  return (
    <>
      <Container locale={props.locale}>
        <>
          <h3>{i18n('bsg.home.intro.title')}</h3>
          <p className="introText">{i18n('bsg.home.intro.text')}</p>
          <div className="categoryLinks">
            {categories.map((category: ICategory) => (
              <Link to={`/investment-compass/${category.link}`}>
                <div className="categoryLink">
                  <div className="iconWrapper">
                    <img
                      className="categoryIcon"
                      src={category.icon}
                      alt={category.title}
                    />
                  </div>
                  <br />
                  <p className="categoryTitle">{category.title}</p>
                </div>
              </Link>
            ))}
          </div>
        </>
        <div style={{ height: 200 }} />
        <div className="disclaimerHome">{i18n('bsg.disclaimer')}</div>
      </Container>
    </>
  );
}

Home.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Home);
