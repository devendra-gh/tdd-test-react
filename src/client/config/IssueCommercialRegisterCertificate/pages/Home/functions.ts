import { IVariables } from '@tamm/app-composer';
import { PATH_SELECT_LICENCE } from '../../routes';
import { addAnalytics, JST_EVENT_KEY } from '../../utils';

const onStart = (props: IVariables) => {
  const {
    history: { push: redirectTo },
  } = props;
  addAnalytics(JST_EVENT_KEY);
  redirectTo(PATH_SELECT_LICENCE);
};

export default { onStart };
