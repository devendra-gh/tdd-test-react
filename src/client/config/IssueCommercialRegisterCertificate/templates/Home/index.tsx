import React from 'react';
import Container from 'client/containers';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import JourneyLandingTemplate from '@tamm/ui-lib-v2-journey-landing-template';
import SidebarTemplate from 'client/templates/Sidebar';
// import { Link } from 'react-router-dom';

/**
 * Home template
 * @param       {Object} props
 * @returns     {JSX}
 */
function Home(props: IVariables) {
  const { i18n, onStart } = props;

  props.actions.breadcrumbs.update(props.breadcrumbs);

  return (
    <>
      <Container
        locale={props.locale}
        sidebar={
          <SidebarTemplate
            i18n={i18n}
            showRelatedJourneyCard
            currentStep=""
            steps={[]}
            stepsStatus={{}}
          />
        }
      >
        <div role="main" aria-label="Home">
          <JourneyLandingTemplate
            title={i18n('service.description.title')}
            description={i18n('service.description.body')}
            startLogin={{
              buttonLabel: i18n('service.startLogin.start'),
              description: i18n('service.startLogin.description'),
              onClick: onStart,
              title: '',
            }}
            process={{
              title: i18n('service.process.title'),
              steps: [
                {
                  description: i18n('service.process.step1.description'),
                  label: i18n('selectLicence'),
                },
                {
                  description: i18n('service.process.step2.description'),
                  label: i18n('makePayment'),
                },
                {
                  description: i18n('service.process.step3.description'),
                  label: i18n('applicationSummary'),
                },
              ],
            }}
            helpfulBlock={{
              callValue: true,
            }}
          />
        </div>
        <div style={{ height: 200 }} />
      </Container>
    </>
  );
}

Home.propTypes = {
  ...routePropTypes,
};

export default withTemplateHooks(Home);
