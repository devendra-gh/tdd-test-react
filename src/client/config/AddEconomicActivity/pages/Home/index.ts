import { IVariables } from '@tamm/app-composer';
import { PATH_HOME } from 'client/config/AddEconomicActivity/routes';
import functions from './functions';

const home = [
  {
    path: PATH_HOME, // path for router
    uniqueId: 'add-economic-activity.home', // uniqueId for caching and other purposes
    template: 'home', // template name, must be located in index of folder template/index
    title: 'addEconomicActivity.title', // title of the page, later it will be read from CMS
    init: functions.init,
    props: {
      title: 'addEconomicActivity.home.title',
      description: 'addEconomicActivity.home.description',
      showRelatedJourneyCard: true,
      startLogin: {
        title: 'addEconomicActivity.title',
        description: 'addEconomicActivity.home.startLogin.description',
        onClick: functions.onClick,
        buttonLabel: 'button.start',
      },
      process: {
        title: 'addEconomicActivity.home.process.title',
        steps: [
          {
            label: 'addEconomicActivity.addActivity',
            description: 'addEconomicActivity.addActivity.description',
          },
          {
            label: 'addEconomicActivity.getDEDApproval',
            description: 'addEconomicActivity.getDEDApproval.description',
          },
        ],
      },
      helpfulBlock: {
        callField: {
          onChange: () => {},
        },
        commentField: {
          onChange: () => {},
        },
        emailField: {
          onChange: () => {},
        },
        onChange: () => {},
        submitButton: {
          onClick: () => {},
        },
        telephoneField: {
          countries: [],
          onChange: () => {},
        },
      },
    },
    state: {
      mapState: [
        'stepsStatus',
        {
          steps: (state: IVariables) => {
            return [];
          },
        },
      ],
      mapDispatch: [
        'stepsStatus',
        'formData',
        'newActivityApiData',
        'helperData',
        'instanceId',
        'businessKey',
      ],
    },
  },
];

export default home;
