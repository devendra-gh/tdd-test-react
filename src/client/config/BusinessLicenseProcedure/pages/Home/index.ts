// import { REQUIRES_LOGIN } from '@tamm/app-composer';
import functions from './functions';

const home = [
  {
    path: ['/business-licence-procedure/'], // path for router
    uniqueId: 'licence-submit', // uniqueId for caching and other purposes
    template: 'businessLicenseProcedureHome', // template name, must be located in index of folder template/index
    title: 'businessLicenseProcedure.main.title', // title of the page, later it will be read from CMS
    props: {
      onStart: functions.onStart,
    },
    state: {
      mapState: [
        'loggedIn',
        'user',
        'instanceId',
        'businessKey',
        {
          steps: () => [],
        },
      ],
      mapDispatch: [
        'instanceId',
        'businessKey',
        'businessLicenseProcedureSteps',
      ],
    },
    // requires: [REQUIRES_LOGIN],
  },
];

export default home;
