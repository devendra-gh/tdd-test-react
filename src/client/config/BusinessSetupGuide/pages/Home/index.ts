import functions from './functions';

const home = [
  {
    path: ['/investment-compass/'], // path for router
    uniqueId: 'home-page', // uniqueId for caching and other purposes
    template: 'home', // template name, must be located in index of folder template/index
    title: 'bsg.home.title', // title of the page, later it will be read from CMS
    init: functions.init,
    onPageInit: functions.onPageInit,
    translations: {
      dictionaryId: 'buy-new-car',
      templateId: '4f3b7eee7185484f988ae2d898d666ba',
    },
    props: {
      breadcrumbs: [
        {
          label: 'bsg.breadcrumb.level0.title',
          link: 'https://www.tamm.abudhabi/',
        },
        {
          label: 'bsg.breadcrumb.level1.title',
          link: 'https://www.tamm.abudhabi/journeys/start-your-business',
        },
      ],
    },
    state: {
      mapState: [],
    },
  },
];

export default home;
