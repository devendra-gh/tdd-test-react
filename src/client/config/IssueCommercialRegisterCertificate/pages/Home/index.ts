import { IVariables } from '@tamm/app-composer';
import functions from './functions';
import { PATH_HOME } from '../../routes';

const home = [
  {
    path: [PATH_HOME], // path for router
    uniqueId: 'home-page', // uniqueId for caching and other purposes
    template: 'home', // template name, must be located in index of folder template/index
    title: 'home.title', // title of the page, later it will be read from CMS
    translations: {
      dictionaryId: 'buy-new-car',
      templateId: '4f3b7eee7185484f988ae2d898d666ba',
    },
    onPageInit: (props: IVariables) => ({
      onStart: () => functions.onStart(props),
    }),
    props: {
      breadcrumbs: [
        {
          label: 'breadcrumb.level0.title',
          link: 'https://www.tamm.abudhabi/',
        },
        {
          label: 'breadcrumb.level1.title',
          link: 'https://www.tamm.abudhabi/tamm-centers-services',
        },
        {
          label: 'breadcrumb.level2.title',
          link:
            'https://www.tamm.abudhabi/tamm-centers-services/department-of-economic-development',
        },
      ],
    },
    state: {
      mapState: [],
    },
  },
];

export default home;
