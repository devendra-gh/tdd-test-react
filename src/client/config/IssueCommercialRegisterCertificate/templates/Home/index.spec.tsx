import React from 'react';
import { shallow } from 'enzyme';
import Home from './index';

jest.mock(
  '@tamm/app-composer/client/templates/withTemplateHooks',
  () => (component: any) => component,
);

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Home Template', () => {
  let props: any;
  beforeEach(() => {
    props = {
      match: {},
      location: {},
      i18n: jest.fn(),
      locale: 'en',
      onStart: jest.fn(),
      breadcrumbs: [],
      actions: {
        breadcrumbs: {
          update: jest.fn(),
        },
      },
    };
  });
  it('should successfully render the home template', () => {
    const component = shallow(<Home {...props} />);
    expect(component).toMatchSnapshot();
  });
});
