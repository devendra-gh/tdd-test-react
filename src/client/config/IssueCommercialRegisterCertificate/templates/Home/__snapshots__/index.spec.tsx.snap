// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Template should successfully render the home template 1`] = `
<Fragment>
  <Container
    locale="en"
    sidebar={
      <Sidebar
        currentStep=""
        i18n={
          [MockFunction] {
            "calls": Array [
              Array [
                "service.description.title",
              ],
              Array [
                "service.description.body",
              ],
              Array [
                "service.startLogin.start",
              ],
              Array [
                "service.startLogin.description",
              ],
              Array [
                "service.process.title",
              ],
              Array [
                "service.process.step1.description",
              ],
              Array [
                "selectLicence",
              ],
              Array [
                "service.process.step2.description",
              ],
              Array [
                "makePayment",
              ],
              Array [
                "service.process.step3.description",
              ],
              Array [
                "applicationSummary",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        showRelatedJourneyCard={true}
        steps={Array []}
        stepsStatus={Object {}}
      />
    }
  >
    <div
      aria-label="Home"
      role="main"
    >
      <JourneyLandingTemplate
        description=""
        helpfulBlock={
          Object {
            "callValue": true,
          }
        }
        process={
          Object {
            "steps": Array [
              Object {
                "description": undefined,
                "label": undefined,
              },
              Object {
                "description": undefined,
                "label": undefined,
              },
              Object {
                "description": undefined,
                "label": undefined,
              },
            ],
            "title": undefined,
          }
        }
        startLogin={
          Object {
            "buttonLabel": undefined,
            "description": undefined,
            "onClick": [MockFunction],
            "title": "",
          }
        }
        tables={null}
        title=""
      />
    </div>
    <div
      style={
        Object {
          "height": 200,
        }
      }
    />
  </Container>
</Fragment>
`;
